kontonummer.js

```diff
@@ -404,24 +409,38 @@
             return {
                 is_valid: false,
                 errors: errors,
-                matched_banks: matched_banks
+                matched_banks: matched_banks,
+                has_warnings: Boolean(warnings.length)
             };
         }
 
         return {
             is_valid: is_valid,
             errors: [],
-            matched_banks: matched_banks
+            matched_banks: matched_banks,
+            has_warnings: Boolean(warnings.length)
         };
     };
 
     var validateLength = function(bank, bankNumber) {
         var errors = [];
 
         if (bank.regex.test(bankNumber)) {
-            if (bankNumber.length < bank.lengths.clearing + bank.lengths.account) {
+            let actualBankNumber = bankNumber;
+            let minBankNumberLength =  bank.lengths.clearing + bank.lengths.account;
+            const maxBankNumberLength = bank.lengths.clearing + bank.lengths.account;
+            
+            if(bank.lengths.min_account) {
+                // Since account number is zero padded we need to slice it before trim
```

Comment:
Isn't it that we need to slice it because it contains both clearing number and account number, rather than because it is zero padded?

If we can avoid comments by writing more code with more expressive behaviour and naming, that is to be preferred. In this case we could create a function that removes the zero padding and name it something like `removeNumberZeroPadding`.